# Study v2: close-case matching

Close cases are matched without using fault, penalty, damage, retirement, or finishing position.
This avoids defining similarity with the outcome the study wants to compare.

In [1]:
# ruff: noqa: E402
import json
import os
from pathlib import Path

ROOT = Path.cwd()
if not (ROOT / "pyproject.toml").exists():
    ROOT = ROOT.parent
os.environ.setdefault("MPLCONFIGDIR", str(ROOT / ".jupyter" / "mplconfig"))

import matplotlib.pyplot as plt
import pandas as pd
from IPython.display import Markdown, display

REVIEW = ROOT / "data/manual/study_v2_review_packets/study-v2-review-7cb1b29b5251"
REFERRAL = ROOT / "data/manual/study_v2_referrals/referrals-5d0559ad2878"
CLOCK = ROOT / "data/manual/study_v2_incident_clock/incident-clock-3dc8bb350308"
CONTEXT = ROOT / "data/manual/study_v2_incident_context/incident-context-a1de9ac2c8ea"
CLOSE = ROOT / "data/manual/study_v2_close_cases/close-cases-36f9bc70de82"
DAMAGE = ROOT / "data/manual/study_v2_damage/damage-screening-66381b550583"
LAYERS = ROOT / "data/manual/study_v2_layers/study-v2-layers-a9b8ff776470"
NATIONALITY = ROOT / "data/manual/study_v2_nationality/nationality-diagnostic-2b1b0ffdd961"
GENERATED = ROOT / "reports/generated/study_v2"
GENERATED.mkdir(parents=True, exist_ok=True)

In [2]:
manifest = json.loads((CLOSE / "manifest.json").read_text(encoding="utf-8"))
summary = pd.read_csv(CLOSE / "conduct_neighbor_summary.csv")
edges = pd.read_csv(LAYERS / "close_case_outcome_contrasts.csv")
coverage = pd.DataFrame(
    [{
        "cases": len(summary),
        "cases_with_at_least_five_neighbors": int((summary["neighbor_count"] >= 5).sum()),
        "neighbor_edges": len(edges),
        "edges_with_different_sanction_outcome": int(edges["different_sanction_outcome"].sum()),
    }]
)
display(coverage)
display(edges.sort_values(["distance", "adjudication_instance_id"]).head(20))
assert "sanction_outcome" in manifest["outcome_fields_excluded_from_primary_match"]

,cases,cases_with_at_least_five_neighbors,neighbor_edges,edges_with_different_sanction_outcome
0,346,317,1655,752


,adjudication_instance_id,adjudication_id,incident_id,neighbor_rank,neighbor_adjudication_instance_id,neighbor_adjudication_id,neighbor_incident_id,distance,fully_compared_context_fields,compared_fields,neighbor_sanction_outcome,neighbor_outcome_family,neighbor_penalty_seconds,neighbor_penalty_points,neighbor_grid_places,match_stage,sanction_outcome,different_sanction_outcome,interpretation
0,adjudication-seed-fia-2018-abu-18bc30990ea3-01,adj-b68335d3407c00bc,incident-src-b68335d3407c00bc,1,adjudication-seed-fia-2018-abu-e3bd8959c0f4-01,adj-9a44d3fc12f61d17,incident-src-9a44d3fc12f61d17,0.0,3,first_lap_candidate|wet_track_candidate|safety...,True,time_penalty,5.0,1.0,NaN,pre_review_feasibility,True,False,review_priority_only_pending_context_and_condu...
1,adjudication-seed-fia-2018-abu-18bc30990ea3-01,adj-b68335d3407c00bc,incident-src-b68335d3407c00bc,2,adjudication-seed-fia-2018-aut-6376daf466a5-01,adj-b25c86494efcda3c,incident-src-b25c86494efcda3c,0.0,3,first_lap_candidate|wet_track_candidate|safety...,False,no_further_action,NaN,NaN,NaN,pre_review_feasibility,True,True,review_priority_only_pending_context_and_condu...
2,adjudication-seed-fia-2018-abu-18bc30990ea3-01,adj-b68335d3407c00bc,incident-src-b68335d3407c00bc,3,adjudication-seed-fia-2018-jpn-af0be8505555-01,adj-590d4355a3039791,incident-src-590d4355a3039791,0.0,3,first_lap_candidate|wet_track_candidate|safety...,True,time_penalty,5.0,1.0,NaN,pre_review_feasibility,True,False,review_priority_only_pending_context_and_condu...
3,adjudication-seed-fia-2018-abu-18bc30990ea3-01,adj-b68335d3407c00bc,incident-src-b68335d3407c00bc,4,adjudication-seed-fia-2019-ita-e4746c5c634f-01,adj-c74375d6086c31ed,incident-src-c74375d6086c31ed,0.0,3,first_lap_candidate|wet_track_candidate|safety...,True,time_penalty,5.0,1.0,NaN,pre_review_feasibility,True,False,review_priority_only_pending_context_and_condu...
4,adjudication-seed-fia-2018-abu-18bc30990ea3-01,adj-b68335d3407c00bc,incident-src-b68335d3407c00bc,5,adjudication-seed-fia-2019-mco-b0bed801d3e3-01,adj-1647a94a71c5efa4,incident-src-1647a94a71c5efa4,0.0,3,first_lap_candidate|wet_track_candidate|safety...,True,time_penalty,5.0,1.0,NaN,pre_review_feasibility,True,False,review_priority_only_pending_context_and_condu...
5,adjudication-seed-fia-2018-abu-754824eaae10-01,adj-58eaa3eee308cb80,incident-src-58eaa3eee308cb80,1,adjudication-seed-fia-2018-abu-e3bd8959c0f4-01,adj-9a44d3fc12f61d17,incident-src-9a44d3fc12f61d17,0.0,3,first_lap_candidate|wet_track_candidate|safety...,True,time_penalty,5.0,1.0,NaN,pre_review_feasibility,True,False,review_priority_only_pending_context_and_condu...
6,adjudication-seed-fia-2018-abu-754824eaae10-01,adj-58eaa3eee308cb80,incident-src-58eaa3eee308cb80,2,adjudication-seed-fia-2018-aut-6376daf466a5-01,adj-b25c86494efcda3c,incident-src-b25c86494efcda3c,0.0,3,first_lap_candidate|wet_track_candidate|safety...,False,no_further_action,NaN,NaN,NaN,pre_review_feasibility,True,True,review_priority_only_pending_context_and_condu...
7,adjudication-seed-fia-2018-abu-754824eaae10-01,adj-58eaa3eee308cb80,incident-src-58eaa3eee308cb80,3,adjudication-seed-fia-2018-jpn-af0be8505555-01,adj-590d4355a3039791,incident-src-590d4355a3039791,0.0,3,first_lap_candidate|wet_track_candidate|safety...,True,time_penalty,5.0,1.0,NaN,pre_review_feasibility,True,False,review_priority_only_pending_context_and_condu...
8,adjudication-seed-fia-2018-abu-754824eaae10-01,adj-58eaa3eee308cb80,incident-src-58eaa3eee308cb80,4,adjudication-seed-fia-2019-ita-e4746c5c634f-01,adj-c74375d6086c31ed,incident-src-c74375d6086c31ed,0.0,3,first_lap_candidate|wet_track_candidate|safety...,True,time_penalty,5.0,1.0,NaN,pre_review_feasibility,True,False,review_priority_only_pending_context_and_condu...
9,adjudication-seed-fia-2018-abu-754824eaae10-01,adj-58eaa3eee308cb80,incident-src-58eaa3eee308cb80,5,adjudication-seed-fia-2019-mco-b0bed801d3e3-01,adj-1647a94a71c5efa4,incident-src-1647a94a71c5efa4,0.0,3,first_lap_candidate|wet_track_candidate|safety...,True,time_

There are 1,655 neighbor links, and 317 of 346 cases reach the five-neighbor support rule. Different
outcomes inside a matched pair are review leads, not proof of inconsistent stewarding. The context
fields and the pair itself still need human confirmation.